static/js/tools/map/fetcher/border_geojson.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Fetch Geo Json data for the border of the containing place.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import { Dispatch, useContext, useEffect } from "react";
+
+import { GeoJsonData } from "../../../chart/types";
+import { ChartDataType, ChartStoreAction } from "../chart_store";
+import { Context } from "../context";
+
+// Use non-DP geojsons for drawing borders to avoid mismatched edges between
+// the border and enclosed places
+const BORDER_GEOJSON_PROPERTY = "geoJsonCoordinates";
+
+/**
+ * Fetch geojson data for the border of the containing place and load into
+ * ChartStore.
+ * @param dispatch react redux dispatch for loading data into a ChartStore.
+ */
+export function useFetchBorderGeoJson(
+  dispatch: Dispatch<ChartStoreAction>
+): void {
+  const { placeInfo } = useContext(Context);
+  useEffect(() => {
+    const contextOk =
+      placeInfo.value.enclosingPlace && placeInfo.value.enclosingPlace.dcid;
+    if (!contextOk) {
+      return;
+    }
+    const action: ChartStoreAction = {
+      type: ChartDataType.BORDER_GEO_JSON,
+      error: null,
+      context: {
+        placeInfo: {
+          enclosingPlace: {
+            dcid: placeInfo.value.enclosingPlace.dcid,
+            name: "",
+          },
+          enclosedPlaceType: placeInfo.value.enclosedPlaceType,
+        },
+      },
+    };
+    axios
```

Comment:
maybe only fetch if we need to show borders? & can make `shouldShowBorder` a util function so that we make sure the logic for fetching and rendering border is the same